portal/views/patch_flask_user.py

```diff
@@ -12,16 +12,17 @@
 def patch_make_safe_url(url):
     """Patch flask_user.make_safe_url() to include '?'
 
-    Turns an usafe absolute URL into a safe relative URL by removing
+    Turns an unsafe absolute URL into a safe relative URL by removing
     the scheme and the hostname
     Example:
         make_safe_url('http://hostname/path1/path2?q1=v1&q2=v2#fragment')
         returns: '/path1/path2?q1=v1&q2=v2#fragment
 
     """
     parts = urlsplit(url)
+    no_scheme, no_hostname = u'', u''
```

Comment:
I think you'll have to remove the `u`, the attributes of `parts` are all `str`, not `unicode`

```
>>> url='http://hostname/path1/path2?q1=v1&q2=v2#fragment'
>>> parts = urlsplit(url)
>>> no_scheme, no_hostname = u'', u''
>>> safe_url = urlunsplit((
...     no_scheme,
...     no_hostname,
...     parts.path,
...     parts.query,
...     parts.fragment,
... ))
Traceback (most recent call last):
  File "<stdin>", line 6, in <module>
  File "/home/ivanc/src/portal/shared_services/env/local/lib/python2.7/site-packages/future/backports/urllib/parse.py", line 399, in urlunsplit
    _coerce_args(*components))
  File "/home/ivanc/src/portal/shared_services/env/local/lib/python2.7/site-packages/future/backports/urllib/parse.py", line 115, in _coerce_args
    raise TypeError("Cannot mix str and non-str arguments")
TypeError: Cannot mix str and non-str arguments
>>> type(parts.path)
<type 'str'>
>>> type(no_scheme)
<type 'unicode'>
>>>
```